### ISSUE TITLE
Improve the UI of the Navigation block display section

### ISSUE THREAD

**Issue by javierarce** [WordPress/gutenberg#44024]
The current interface of Navigation block display section is quite convoluted. 

Because the preview sits on top of the section, if the user selects "Mobile" or "Always" in the segment control and then expands the preview, the main setting gets pushed to the bottom, making the UI confusing. 

![image](https://user-images.githubusercontent.com/4933/189329736-eb96cebe-98bb-4dfa-8569-730644f90af4.png)

<details>
<summary>See initial proposal</summary>

I propose we switch the order of the setting and tweak the height of the preview and the size of the X icon:

![image](https://user-images.githubusercontent.com/4933/189330164-e2f1c7fd-3eaa-44ea-8a2e-e290e6dc7e26.png)
</details>

This enhancement will also help us address [this feature request](https://github.com/WordPress/gutenberg/issues/40580) to customize the labels:

----

### Proposed design for implementation (2022-09-13)

![image](https://user-images.githubusercontent.com/4933/195557481-29b592d1-251f-4969-87f3-b92a3fdb5a36.png)
[Design](https://www.figma.com/file/mlOogoixVzCoGc4O5Qos1j/Navigation-block-improvements?node-id=953%3A24854) - [Prototype](https://www.figma.com/proto/mlOogoixVzCoGc4O5Qos1j/Navigation-block-improvements?page-id=953%3A24854&node-id=961%3A25163&viewport=-194%2C-183%2C0.18&scaling=min-zoom&starting-point-node-id=961%3A25163)

#### Proposed position on the sidebar

![image](https://user-images.githubusercontent.com/4933/196378076-48f9564c-5652-4530-8394-a25844395e4b.png)

**Comment by mtias** [comment]
I think the preview being an interactive element is a bit of an obscure interaction, so improving that would be useful. Instead of a toggle for "Use icons" can we try another segmented control that says "Icon / Text" that toggles between the options? I'm not sure we should allow picking the menu and close set ups separately, it seems to add a bunch of cognitive complexity.

Also, when the label is set, we should probably allow further customization (typography, border, etc).

**Comment by javierarce** [comment]
I agree with you about the preview being problematic. Here's an iteration based on your feedback:

![image](https://user-images.githubusercontent.com/4933/189618145-987904db-7c37-407a-a1aa-ea71966893cb.png)

- The preview is expanded using a simple toggle
- We use a segment control to choose between icon and text modes
- The segment control acts as the labels for the components below
- [Prototype](https://www.figma.com/proto/mlOogoixVzCoGc4O5Qos1j/Navigation-block-improvements?page-id=953%3A24854&node-id=991%3A27204&viewport=742%2C-412%2C0.13&scaling=min-zoom&starting-point-node-id=991%3A27204&show-proto-sidebar=1)

> Also, when the label is set, we should probably allow further customization (typography, border, etc).

Since we have typography and color settings in other sections of this block, I understand you mean that you want to allow customizing these labels independently from the rest of the menu, right? But wouldn't that make things more complicated?

**Comment by javierarce** [comment]
Added a final iteration based on @jasmussen suggestion: instead of a toggle, use the settings icon as we do in other sections.


![image](https://user-images.githubusercontent.com/4933/189852144-213d9ddf-9e3a-4aaf-bf47-1dbbaa139a8b.png)

**Comment by jasmussen** [comment]
Wanted to note that for now we've removed the two ellipses from the icon options. (Let's explore a couple of excellent replacements, not necessarily as part of this PR, just noting)

**Comment by javierarce** [comment]
Thanks, @jasmussen, I've updated the design in Figma and in the issue description to reflect that change.

**Comment by javierarce** [comment]
I updated the original issue above to show the proposed position of this section in the sidebar: currently, we show the Display section under Layout and Menu, but it should appear above Layout since Display defines the basic appearance and behaviour of the navigation.

**Comment by getdave** [comment]
@aaronrobertshaw Are you aware of a way we can force the `Display` section to appear above the `Layout` section? I know you worked a lot on InspectorControls grouping and tabs. Any help appreciated 🙇‍♂️

**Comment by getdave** [comment]
@SaxonF It would be worth adding this to tasks to complete in 6.3 - are you happy with the design proposed above?

Reorganising this sidebar a bit may well improve user interactions with key tasks. Currently it feels way too focused on advanced controls such as layout where I expect that 80% of user interactions will be for:

- color
- overlay menu settings

**Comment by SaxonF** [comment]
@getdave Looks good to me. I'd personally like to see layout grouped with dimensions, which [this issue](https://github.com/WordPress/gutenberg/issues/47902#issuecomment-1478188344) looks at, so the positioning question might not be an issue anymore.

**Comment by aaronrobertshaw** [comment]
Thanks for the ping @getdave 👍 

> Are you aware of a way we can force the Display section to appear above the Layout section?

Both the Nav block's Display panel, and the Layout block support panel, render to the `default` group for the `InspectorControls`. There isn't currently any means to order that slot's fills. As the Layout panel is from block supports, I believe that gets added reliably to the fills first, hence its current position.

There could be a couple of options here but none that came to mind were particularly clean or elegant.

1. Add a new `InspectorControls` group that can be rendered before the default group.
2. Add a new `InspectorControls` group for `layout` which can be added after the default group.

Both of those options, leave us open to plugins abusing the top group's slot for promoting their controls beyond the prominence they warrant.

Assuming we can find a means to restrict the use of a new group to only the desired blocks, and only core controls, then option `#1` might work.

I'm not sure of its relevance here but there was an old issue (https://github.com/WordPress/gutenberg/issues/35541) that evaluated the order of block support panels in the sidebar. So the current order was at least deliberate.

> I'd personally like to see layout grouped with dimensions, which https://github.com/WordPress/gutenberg/issues/47902#issuecomment-1478188344 looks at, so the positioning question might not be an issue anymore.

If Layout was merged with Dimensions and that panel relocated to the Settings tab, we'd avoid the need for a `layout` group but still face the same decision around allowing the block's default controls (and therefore plugins') above the layout supports.

**Comment by jasmussen** [comment]
> Looks good to me. I'd personally like to see layout grouped with dimensions, which https://github.com/WordPress/gutenberg/issues/47902#issuecomment-1478188344 looks at, so the positioning question might not be an issue anymore.

I've not had a chance to form deep thoughts here, but if we do consolidate dimension controls in the layout panel, it needs a somewhat solid mockup that considers all the use cases, including how dimensions change depending on the [layout type](https://github.com/WordPress/gutenberg/issues/38022#issuecomment-1479061977).

**Comment by getdave** [comment]
Thanks Aaron 👍 

> Assuming we can find a means to restrict the use of a new group to only the desired blocks, and only core controls, then option `#1` might work.

Yes it feels like Core should be able to decide how it wants to order the panels. How to do that....is another question 😄

**Comment by aaronrobertshaw** [comment]
> How to do that....is another question 

I haven't looked into it but it would be nice if we could use the [lock/unlock feature](https://github.com/WordPress/gutenberg/pull/46131) to make the new InspectorControls group "private" or "core-only".
